components/skus/browser/skus_url_loader_impl_unittest.cc

```diff
@@ -70,3 +96,24 @@ TEST_F(SkusUrlLoaderImplUnitTest, SanitizedResponse) {
   SetResponseText("{,}");
   EXPECT_TRUE(GetRequestResponse("GET", "https://brave.com").is_none());
 }
+
+TEST_F(SkusUrlLoaderImplUnitTest, BeginFetch) {
+  SetResponseText("{}");
+  FetchResponse(
+      "GET",
+      rust::cxxbridge1::Box<skus::HttpRoundtripContext>::from_raw(nullptr),
+      "https://brave.com", skus::SkusResult::Ok, "{}");
+
+  SetResponseText("");
+  FetchResponse(
+      "GET",
+      rust::cxxbridge1::Box<skus::HttpRoundtripContext>::from_raw(nullptr),
+      "https://brave.com", skus::SkusResult::Ok, "");
+
+  SetResponseText("");
+  SetResponseCode(net::HTTP_INTERNAL_SERVER_ERROR);
+  FetchResponse(
+      "GET",
+      rust::cxxbridge1::Box<skus::HttpRoundtripContext>::from_raw(nullptr),
+      "https://brave.com", skus::SkusResult::Ok, std::string());
```

Comment:
@husobee it looks weird to return Ok for 5xx codes, it happens because of `bool success = api_request_result.IsResponseCodeValid();` where 
```
bool APIRequestResult::IsResponseCodeValid() const {
  return response_code_ >= 100 && response_code_ <= 599;
}
```
https://github.com/brave/brave-core/commit/d46fcb546cd2844ef4555b5d5e8f24c78b3090e1#diff-67e03ef3ac9634121f3f855719a22a33fbf18bf458dafbdc7761e35b26ff7bfcR74

do we actually need this? maybe we can pass code into the Rust lib and skip errors inside? wdyt?

Reply:
This specifically allows the rust skus sdk library to figure out on its own how to handle any "valid" response code including the internal server error types of response codes.  Within the skus sdk we will handle the response code appropriately.  In this particular case the "success" just means that we got a response at all.  Success=false is partiularly for connection timeouts, or other non application level error states.

Reply:
I see, thanks!